memory/arena.h

```diff
@@ -78,7 +78,7 @@ class Arena : public Allocator {
   size_t BlockSize() const override { return kBlockSize; }
 
   bool IsInInlineBlock() const {
-    return blocks_.empty();
+    return blocks_.empty() && huge_blocks_.empty();
```

Comment:
I do not think this is necessarily enough.  It is possible that huge_blocks_ is not empty but an allocation from huge blocks failed, leaving "empty" elements in the huge_blocks_ array,

Reply:
I didn't invetigated the arena implementation in detail, I just changed this line, then the failed UT passed. Maybe there are better fix for this issue.